influxdb_connector/influxdb_connector.py

```diff
@@ -0,0 +1,257 @@
+import os
+import argparse
+import pandas as pd
+from collections import defaultdict
+from cnvrgv2 import Cnvrg
+from influxdb_client import InfluxDBClient
+from influxdb_client.client.write_api import SYNCHRONOUS
+
+cnvrg_workdir = os.environ.get("CNVRG_WORKDIR", "/cnvrg")
+
+
+def parse_parameters():
+    """Command line parser."""
+    parser = argparse.ArgumentParser(description="""influxdb Connector""")
+    parser.add_argument(
+        "--token",
+        action="store",
+        dest="token",
+        required=True,
+        help="""--- influxdb API Access Token ---""",
+    )
+    parser.add_argument(
+        "--url",
+        action="store",
+        dest="url",
+        required=True,
+        help="""--- influxdb access url ---""",
+    )
+    parser.add_argument(
+        "--org",
+        action="store",
+        dest="org",
+        required=True,
+        help="""--- influxdb access organization account ---""",
+    )
+    parser.add_argument(
+        "--bucket",
+        action="store",
+        dest="bucket",
+        required=True,
+        help="""--- bucket where the data is pulled from ---""",
+    )
+    parser.add_argument(
+        "--measurement",
+        action="store",
+        dest="measurement",
+        required=True,
+        help="""--- measurement name where the data is pulled from ---""",
+    )
+    parser.add_argument(
+        "--time_col",
+        action="store",
+        dest="time_col",
+        required=True,
+        help="""--- name of the time column, which will be used for indexing ---""",
+    )
+    parser.add_argument(
+        "--range_start",
+        action="store",
+        dest="range_start",
+        required=False,
+        default="-10y",
+        help="""--- fetch from starting datetime in the format of 2020-01-01T00:00:00Z, or specify -1d, -1h, -1m for last day, hour, or minute ---""",
+    )
+    parser.add_argument(
+        "--range_end",
+        action="store",
+        dest="range_end",
+        required=False,
+        default="now()",
+        help="""--- fetch through ending datetime in the format of 2020-01-01T00:00:00Z, or defaults to now() for current time ---""",
+    )
+    parser.add_argument(
+        "--local_dir",
+        action="store",
+        dest="local_dir",
+        required=False,
+        default=cnvrg_workdir,
+        help="""--- The path to save the dataset file to ---""",
+    )
+    parser.add_argument(
+        "--cnvrg_dataset",
+        action="store",
+        dest="cnvrg_dataset",
+        required=False,
+        default="None",
+        help="""--- the name of the cnvrg dataset to store in ---""",
+    )
+    parser.add_argument(
+        "--file_name",
+        action="store",
+        dest="file_name",
+        required=False,
+        default="influxdb.csv",
+        help="""--- name of the dataset csv file ---""",
+    )
+    return parser.parse_args()
+
+
+# custom exceptions
+class NoneCnvrgDatasetError(Exception):
```

Comment:
Not sure if you are still using this custom exception. You can probably delete this or raise it somewhere in the cnvrg dataset snippet.

Reply:
You're right, NoneCnvrgDatasetError is no longer used and deleted.